independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/codestarts/quarkus/QuarkusCodestartCatalog.java

```diff
@@ -165,6 +175,51 @@ public static QuarkusCodestartCatalog fromExtensionsCatalog(ExtensionCatalog cat
         return projectCodestarts;
     }
 
+    private Optional<ExtensionCodestart> getSelectedDefaultCodeStart(QuarkusCodestartProjectInput projectInput) {
+        // This is very hackyish, we need a better data structure to do better
+        Optional<ArtifactCoords> quarkusBom = projectInput.getBoms().stream()
+                .map(ArtifactCoords::fromString)
+                .filter(b -> isCoreBom(b) || isPlatformBom(b) || isUniverseBom(b))
+                .findFirst();
+
+        String bomVersion = null;
+
+        if (quarkusBom.isPresent()) {
+            bomVersion = quarkusBom.get().getVersion();
+        }
+
+        if (bomVersion == null || VersionScheme.MAVEN.compare(bomVersion, "2.8") >= 0) {
+            if (projectInput.getExtensions().isEmpty() ||
+                    (projectInput.getExtensions().size() == 1
+                            && isLanguageExtension(projectInput.getExtensions().iterator().next()))) {
+                return Optional.of(ExtensionCodestart.RESTEASY_REACTIVE);
+            }
+
+            return Optional.empty();
+        }
+
+        return Optional.of(ExtensionCodestart.RESTEASY);
+    }
+
+    private boolean isCoreBom(ArtifactCoords artifactCoords) {
+        return IO_QUARKUS_GROUP_ID.equals(artifactCoords.getGroupId()) && QUARKUS_BOM.equals(artifactCoords.getArtifactId());
+    }
+
+    private boolean isUniverseBom(ArtifactCoords artifactCoords) {
+        return IO_QUARKUS_GROUP_ID.equals(artifactCoords.getGroupId())
+                && QUARKUS_UNIVERSE_BOM.equals(artifactCoords.getArtifactId());
+    }
+
+    private boolean isPlatformBom(ArtifactCoords artifactCoords) {
+        return IO_QUARKUS_PLATFORM_GROUP_ID.equals(artifactCoords.getGroupId())
```

Comment:
This is where you could check for `com.redhat.quarkus.platform`

Reply:
I added that.